v4.9.0+_MIGRATION_GUIDE.md

```diff
@@ -0,0 +1,1019 @@
+    android:exported="true">
+    ...
+    <intent-filter>
+        <action android:name="android.intent.action.VIEW"/>
+        <data android:scheme="custom-url-scheme-2"/>
+        <category android:name="android.intent.category.DEFAULT"/>
+        <category android:name="android.intent.category.BROWSABLE"/>
+    </intent-filter>
+</activity>
+``` 
+
+Then when constructing your `BraintreeClient` make sure to pass the appropriate custom url scheme for each deep link target Activity:
+
+```java
+// MyPaymentsActivity1.java
+BraintreeClient braintreeClient =
+    new BraintreeClient(this, "TOKENIZATION_KEY_OR_CLIENT_TOKEN", "custom-url-scheme-1");
+```
+
+```java
+// MyPaymentsActivity2.java
+BraintreeClient braintreeClient =
+    new BraintreeClient(this, "TOKENIZATION_KEY_OR_CLIENT_TOKEN", "custom-url-scheme-2");
+```
+
+## BraintreeFragment
+
+In v4, we decoupled the Braintree SDK from Android to offer more integration flexibility. 
+`BraintreeFragment` has been replaced by a `Client` for each respective payment feature. 
+See the below payment method sections for examples of instantiating and using the feature clients. 
+
+## Authorization Provider
+
+When creating a `BraintreeClient`, you can provide a tokenization key, client token, or an `AuthorizationProvider`. When given an authorization provider, the SDK will fetch a client token on your behalf when it is needed. This makes it possible to construct a `BraintreeClient` instance in `onCreate`.
+
+**Assuming** you have a server that supports GET `https://www.my-api.com/client_token` and receives the following json response:
+
+```json
+{
+  "value": "<CLIENT_TOKEN>"
+}
+```
+
+Here is an example `AuthorizationProvider` implementation using Retrofit 1.x:
+
+```java
+// ClientToken.java
+public class ClientToken {
+
+  @SerializedName
+  private String value;
+
+  public String getValue() {
+    return value;
+  }
+}
+
+// Api.java
+public interface Api {
+
+  @GET("/client_token")
+  void getClientToken(Callback<ClientToken> callback);
+}
+
+// ExampleAuthorizationProvider.java
+class ExampleAuthorizationProvider implements AuthorizationProvider {
+
+    private Api api = new RestAdapter.Builder()
+            .setEndpoint("https://my-api.com")
+            .build()
+            .create(Api.class);
+
+  void getClientToken(@NonNull ClientTokenCallback callback) {
+    api.getClientToken(new Callback<ClientToken>() {
+      
+      @Override
+      public void success(ClientToken clientToken, Response retrofitResponse) {
+        if (retrofitResponse.getStatus() == 200) {
+          callback.onSuccess(clientToken.getValue());
+        } else {
+          callback.onFailure(new Exception("An unknown error occurred."));
+        }
+      }
+
+      @Override
+      public void failure(RetrofitError error) {
+        // forward error to the SDK
+        callback.onFailure(error);
+      }
+    });
+  }
+}
+```
+
+Then in an Activity or Fragment, create an instance of `BraintreeClient`:
```

Comment:
Can this be done in a Fragment? Originally said Activity, but in our example code snippets we were doing this in a fragment previously

Reply:
It is possible but we never actually made overloads for `Fragment`. I wonder if we should to simplify the integration pattern?